### DISCUSSION TITLE
Database service - what is acquire method for?

### DISCUSSION THREAD

**Discussion by ThePlenkov** [cap-js/.github#2] [General]
Hi!

I've started a new database service inheriting from cds.Database service. Currently my get requests get failed here:

![image](https://github.com/cap-js/.github/assets/6381507/66e05c81-39c1-4f8a-9011-b71529b8da6c)

```
  async begin() {
    const tx = this.context ? this : this.tx()
    tx. = await tx.acquire(tx.context)
    try {
      await tx.send('BEGIN')
    } catch (e) {
      tx.release(tx.dbc)
      throw e
    }
    return tx
  }
```

so what happens - the default class logic is like this - in case if context is provided ( like Read Entity ) then it tries to call acquire method of itself. There are several issues:
- there is no default acquire method implementation. I see that Hana and SQlite classes have this method declared directly there
-  for TS experience I propose to define this method also in the parent class even as an abstract one. It's not yet clear also - what should be returned from this method too

Thanks!

**Comment by ThePlenkov** [comment]
@schwma do you know who can help? Thanks

**Comment by schwma** [comment]
@patricebender @BobdenOs @SamuelBrucksch could you please take a look?

**Comment by BobdenOs** [comment]
@ThePlenkov `cds.DatabaseService` is the old database service. You should inherit from the pluggable database service which can be found [here](https://github.com/cap-js/cds-dbs/blob/main/db-service/lib/common/DatabaseService.js#L11)

If your target database uses `SQL` you can inherit from the `SQLService` directly. Which can be found [here](https://github.com/cap-js/cds-dbs/blob/main/db-service/lib/SQLService.js#L7).

**Reply by ThePlenkov** [reply]
Thanks for the reply. The principal difference - is that community project doesn't have typescript yet. So it's hard to use it without definitions. So my idea was to create another class similar to cds-dbs/DatabaseService. However the initital question was about acquire -  it looks like this method shouldn't be used by redefining [begin](https://github.com/cap-js/cds-dbs/blob/main/db-service/lib/common/DatabaseService.js#L44) method in a common class.

**Reply by BobdenOs** [reply]
The new database service doesn't redefine begin. It inherits from cds.Service not from cds.DatabaseService. It explicitly defines begin so that all depending classes have a structured API to implement. By inheriting from the pluggable db service. It is only needed to define a factory and the set function. As is hinted at by the error message `2b overriden in subclass`.

As for typescript definitions of the db-service package. The whole API is still in beta and for some parts there are already jsdoc definitions provided, but not yet for everything.
